dapp_runner/runner/runner.py

```diff
@@ -249,7 +259,57 @@ async def _listen_state_queue(self, service: DappService):
                 return
 
             # on a state change, we're publishing the state of the whole dapp
-            self.state_queue.put_nowait(self.dapp_state)
+            self._report_status_change()
+
+    def _report_status_change(self) -> None:
+        """Emit message with full state update to state queue."""
+
+        nodes_states = self.dapp_state
+
+        self.state_queue.put_nowait(
+            {
+                "nodes": nodes_states,
+                "app": self._get_app_state_from_nodes(
+                    len(self.dapp.nodes), self._desired_app_state, nodes_states
+                ),
+                "timestamp": utcnow_iso_str(),
+            }
+        )
+
+    # FIXME: Usage of @staticmethod is only for sake of unittests - to be removed
+    @staticmethod
+    def _get_app_state_from_nodes(
+        dapp_node_count: int,
+        desired_app_state: ServiceState,
+        nodes_states: Dict[str, Dict[int, ServiceState]],
+    ) -> ServiceState:
```

Comment:
again, I strongly believe that the state of the whole app should not be tracked using the same state class as the state of each instance of a single service.... e.g. I imagine we could have a state of `updating` when we change the structure of the app and the services would need to be started/stopped according to the plan but the app as a whole would neither be "starting" nor "stopping"...